### PR TITLE
Issue 253: Fix linting

### LINKED ISSUE #253: Linting only runs on two files!
Whoops:

image

Turns out we are linting just **two** posts, because the glob pattern in lintPosts.js is this:
`"*/_posts/**/*.{md,html}"`

The pattern should be
`"_posts/**/*.{md,markdown,html}"`

And with that, predictably, we see a colossal amount of errors! However, all is not lost:
- most of these are down to "more than one category" chosen for a post, which is not allowed (_Why is that?_), though the vast majority of these seem to have the author name as an extra category
- two are "does not have a recognised category" due to the author using "AI" instead of "Artificial Intelligence"
- the remaining ones (25) appear to be due to another piece of logic expecting posts to be within author subdirectories, and once that is corrected, these all become "summary is less than 130 characters". Is 130 characters a meaningful limit? It seems relatively common that authors have less than this.

### PR DESCRIPTION
**Pull request by dogle-scottlogic**
Fix the linting, addressed issue #253 :
- Update path matching to match all posts correctly
- Remove author from category (Note: this seems to be something siteleaf adds so might need to look at that)
- Allow up to 3 categories to be specified
- Make 130 char summary a soft (WARN) min
- Fix unknown category issues

### CODE REVIEW

**Comment by dogle-scottlogic** [comment]
As far as I can see the accessibility job is failing due to the number of files changed - although the feedback isn't super helpful

**Comment by chriswilty** [comment]
There are a _lot_ of whitespace changes, moving summary lines onto a single line. Surely it's not the linter does that? It is making this PR far larger than it needs to be.

Also, I'm not convinced Siteleaf adds author as category, as my first post from 9th May was done in siteleaf and is not in the change list.

**Comment by dogle-scottlogic** [comment]
This was a side effect of the method I used to remove the author. I've tried a different approach which seems to preserve the formatting of other fields

File: _posts/2023-10-26-conscientious-computing-facing-into-big-tech-challenges.markdown [hunk 1]
```diff
@@ -6,7 +6,6 @@ categories:
 tags:
 - sustainable software
 - Sustainability
-- ocronk
```
Comment (chriswilty) [inline _posts/2023-10-26-conscientious-computing-facing-into-big-tech-challenges.markdown]: Strictly speaking, this (and a few others I've seen) are ok as they are Tags, not Categories.

**Review by chriswilty — APPROVED** [review]
I've not checked every single file, but I have noted that most are single-line removals which makes sens. The linter script also seems fine.

Don't worry about the pa11y error, we're not changing any actual post content so can ignore it.

**Pull request merged**